src/main/java/seedu/address/logic/newcommands/CreateDeadlineCommand.java

```diff
@@ -0,0 +1,139 @@
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.newcommands.exceptions.CommandException;
+import seedu.address.model.id.GroupId;
+import seedu.address.model.id.StudentId;
+import seedu.address.model.id.exceptions.InvalidIdException;
+import seedu.address.model.path.AbsolutePath;
+import seedu.address.model.path.RelativePath;
+import seedu.address.model.path.exceptions.InvalidPathException;
+import seedu.address.model.path.exceptions.UnsupportedPathOperationException;
+import seedu.address.model.profbook.Group;
+import seedu.address.model.profbook.Root;
+import seedu.address.model.profbook.Student;
+import seedu.address.model.statemanager.GroupOperation;
+import seedu.address.model.statemanager.RootOperation;
+import seedu.address.model.statemanager.StudentOperation;
+import seedu.address.model.taskmanager.Deadline;
+import seedu.address.model.taskmanager.exceptions.NoSuchTaskException;
+
+/**
+ * Adds a Deadline for a specified {@code Student} or {@code Group}.
+ */
+public class CreateDeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+
+    public static final String MESSAGE_SUCCESS = "New Deadline task added: %1$s";
+    public static final String MESSAGE_DUPLICATE_DEADLINE_TASK =
+            "This Deadline task has already been allocated";
+
+    public static final String MESSAGE_INCORRECT_DIRECTORY_ERROR = "Directory is invalid";
+    protected AbsolutePath absolutePath;
+    protected Student stu;
+    protected Group grp;
+    private final RelativePath path;
+    private final Deadline deadline;
+
+    /**
+     * Creates an CreateDeadlineCommand to add the Deadline Task for a specified {@code Student} or {@code Group}
+     */
+    public CreateDeadlineCommand(RelativePath path, Deadline deadline) {
+        requireAllNonNull(path, deadline);
+        this.path = path;
+        this.deadline = deadline;
+    }
+
+    /**
+     * Executes an CreateDeadlineCommand to allocate a {@code Deadline} task to a {@code Group} or {@code Student}
+     *
+     * @param currPath The current path user is at in ProfBook.
+     * @param root The root in ProfBook.
+     * @return Command result which represents the outcome of the command execution.
+     * @throws CommandException Exception thrown when error occurs during command execution.
+     * @throws InvalidPathException thrown when error occurs due to invalid path.
+     * @throws UnsupportedPathOperationException Exception thrown when error occurs due to unsupported path execution.
+     * @throws InvalidIdException Exception thrown due to invalid Id which includes GroupId and StudentId.
+     * @throws NoSuchTaskException Exception thrown due to invalid Task.
+     */
+    @Override
+    public CommandResult execute(AbsolutePath currPath, Root root) throws CommandException,
+            InvalidPathException, UnsupportedPathOperationException, InvalidIdException, NoSuchTaskException {
```

Comment:
CommandResult should only throw CommandException as the caller only catch the CommandException and show the error message on popup. Please catch other exceptions in the execute command and rethrow as CommandException with user friendly message. Same issues found for other command class.